src/types/index.ts

```diff
@@ -369,6 +369,9 @@ export const AssertionSchema = z.object({
   // The expected value, if applicable
   value: z.custom<AssertionValue>().optional(),
 
+  // An external object that is passed to the assertion for custom javascript asserts
+  config: z.any().optional(),
```

Comment:
I think this should be `Record<string, any>`!

Reply:
So I was chatting about this with @mldangelo and it theoretically could be an object of arbitrary structure (e.g. keys don't necessarily need to be strings here). Happy to change it just be a string -> any mapping if you think that makes more sense.